analysis/report/plot_measures_report.py

```diff
@@ -28,6 +38,236 @@ def group_medications(medications_df):
     return pd.concat(by_line)
 
 
+# NOTE: bug with pandas 1.01, cannot do pivot with MultiIndex
+def MultiIndex_pivot(
+    df: pd.DataFrame,
+    index: str = None,
+    columns: str = None,
+    values: str = None,
+) -> pd.DataFrame:
+    """
+    https://github.com/pandas-dev/pandas/issues/23955
+    Usage:
+    df.pipe(MultiIndex_pivot, index = ['idx_column1', 'idx_column2'],
+            columns = ['col_column1', 'col_column2'], values = 'bar')
+    """
+    output_df = df.copy(deep=True)
+    if index is None:
+        names = list(output_df.index.names)
+        output_df.reset_index(drop=True, inplace=True)
+    else:
+        names = index
+    output_df = output_df.assign(
+        tuples_index=[tuple(i) for i in output_df[names].values]
+    )
+    if isinstance(columns, list):
+        output_df = output_df.assign(
+            tuples_columns=[tuple(i) for i in output_df[columns].values]
+        )
+        output_df = output_df.pivot(
+            index="tuples_index", columns="tuples_columns", values=values
+        )
+        output_df.columns = pd.MultiIndex.from_tuples(
+            [((x[0],) + x[1]) for x in output_df.columns],
+            names=[None] + columns,
+        )
+    else:
+        output_df = output_df.pivot(
+            index="tuples_index", columns=columns, values=values
+        )
+    output_df.index = pd.MultiIndex.from_tuples(output_df.index, names=names)
+    return output_df
+
+
+def produce_min_max_table(df, column_to_plot):
+    table = df.pipe(
+        MultiIndex_pivot,
+        index=["name", "type"],
+        columns="gas_year",
+        values=["numerator", "rate"],
+    )
+    if column_to_plot == "numerator":
+        return table["numerator"]
+    else:
+        table.numerator = table.numerator.astype(float)
+        table.rate = table.rate.astype(float)
+        ratio = table["rate"]["2023"] / table["rate"]["2018"]
+
+        # See formula of ci irr:
```

Comment:
This one is a bit hard since its currently using a strange dataframe with a multiindex in the rows and the columns. I have updated the ci of a proportion function and conversion to string to be as reusable and general as possible (as it takes as input the joined measure file). 
In a future round I'll see if I can make this more general.